File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/ftrl/medlemskapsperiode/dto/MedlemskapsperiodeDto.java
package no.nav.melosys.tjenester.gui.ftrl.medlemskapsperiode.dto;

import java.time.LocalDate;

import no.nav.melosys.domain.Medlemskapsperiode;
import no.nav.melosys.domain.kodeverk.Bestemmelse;
import no.nav.melosys.domain.kodeverk.InnvilgelsesResultat;
import no.nav.melosys.domain.kodeverk.Medlemskapstyper;
import no.nav.melosys.domain.kodeverk.Trygdedekninger;

public record MedlemskapsperiodeDto(long id,
                                    LocalDate fomDato,
                                    LocalDate tomDato,
                                    Bestemmelse bestemmelse,
                                    InnvilgelsesResultat innvilgelsesResultat,
                                    Trygdedekninger trygdedekning,
                                    Medlemskapstyper medlemskapstype) {

    public static MedlemskapsperiodeDto av(Medlemskapsperiode medlemskapsperiode) {
        return new MedlemskapsperiodeDto(
            medlemskapsperiode.getId(),
            medlemskapsperiode.getFom(),
            medlemskapsperiode.getTom(),
            medlemskapsperiode.getBestemmelse(),
            medlemskapsperiode.getInnvilgelsesresultat(),
            medlemskapsperiode.getTrygdedekning(),
            medlemskapsperiode.getMedlemskapstype()
        );
    }
}

File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/ftrl/medlemskapsperiode/dto/MedlemskapsperiodeDto.kt
package no.nav.melosys.tjenester.gui.ftrl.medlemskapsperiode.dto

import no.nav.melosys.domain.Medlemskapsperiode
import no.nav.melosys.domain.kodeverk.Bestemmelse
import no.nav.melosys.domain.kodeverk.InnvilgelsesResultat
import no.nav.melosys.domain.kodeverk.Medlemskapstyper
import no.nav.melosys.domain.kodeverk.Trygdedekninger
import java.time.LocalDate

data class MedlemskapsperiodeDto(
    val id: Long,
    val fomDato: LocalDate,
    val tomDato: LocalDate,
    val bestemmelse: Bestemmelse,
    val innvilgelsesResultat: InnvilgelsesResultat,
    val trygdedekning: Trygdedekninger,
    val medlemskapstype: Medlemskapstyper
) {
    companion object {
        fun av(medlemskapsperiode: Medlemskapsperiode): MedlemskapsperiodeDto {
            return MedlemskapsperiodeDto(
                medlemskapsperiode.id,
                medlemskapsperiode.fom,
                medlemskapsperiode.tom,
                medlemskapsperiode.bestemmelse,
                medlemskapsperiode.innvilgelsesresultat,
                medlemskapsperiode.trygdedekning,
                medlemskapsperiode.medlemskapstype
            )
        }
    }
}
